test/stress_tests.py

```diff
@@ -108,5 +108,145 @@ def g(x):
+                                                 redis_address,
+                                                 objstore_memory=self.plasma_store_memory)
+    address_info = {
+        "redis_address": redis_address,
+        "object_store_addresses": [plasma_address],
+        }
+
+    # Start the rest of the services in the Ray cluster.
+    ray.worker._init(address_info=address_info, start_ray_local=True,
+                     num_workers=1)
+
+  def tearDown(self):
+    # Clean up the Ray cluster.
+    self.assertTrue(ray.services.all_processes_alive())
+    ray.worker.cleanup()
+
+  def testSingleNodeSimple(self):
+    # Define the size of one task's return argument so that the combined sum of
+    # all objects' sizes is twice the plasma store's allotted memory.
+    num_objects = 1000
+    size = int(self.plasma_store_memory * 2 / (num_objects * 8))
+
+    # Define a remote task with no dependencies, which returns a numpy array of
+    # the given size.
+    @ray.remote
+    def foo(i, size):
+      array = np.zeros(size)
+      array[0] = i
+      return array
+
+    # Launch num_objects instances of the remote task.
+    args = []
+    for i in range(num_objects):
+      args.append(foo.remote(i, size))
+
+    # Get each value to force each task to finish. After some number of gets,
+    # old values should be evicted.
+    for i in range(num_objects):
+      value = ray.get(args[i])
+      self.assertEqual(value[0], i)
+    # Get each value again to force reconstruction.
+    for i in range(num_objects):
+      value = ray.get(args[i])
+      self.assertEqual(value[0], i)
+
+  def testSingleNodeRecursive(self):
+    # Define the size of one task's return argument so that the combined sum of
+    # all objects' sizes is twice the plasma store's allotted memory.
+    num_iterations = 1000
+    size = int(self.plasma_store_memory * 2 / (num_iterations * 8))
+
+    # Define a root task with no dependencies, which returns a numpy array of
+    # the given size.
+    @ray.remote
+    def no_dependency_task(size):
+      array = np.zeros(size)
+      return array
+
+    # Define a task with a single dependency, which returns its one argument.
+    @ray.remote
+    def single_dependency(i, arg):
+      arg = np.copy(arg)
+      arg[0] = i
+      return arg
+
+    # Launch num_iterations instances of the remote task, each dependent on the
+    # one before it.
+    arg = no_dependency_task.remote(size)
+    args = []
+    for i in range(num_iterations):
+      arg = single_dependency.remote(i, arg)
+      args.append(arg)
+
+    # Get each value to force each task to finish. After some number of gets,
+    # old values should be evicted.
+    for i in range(num_iterations):
+      value = ray.get(args[i])
+      self.assertEqual(value[0], i)
+    # Get each value again to force reconstruction.
+    for i in range(num_iterations):
+      value = ray.get(args[i])
+      self.assertEqual(value[0], i)
+
```

Comment:
Can we either add another pass through in the reverse direction (in addition to what you have)? E.g.,

```python
for i in range(num_iterations)[::-1]:
  value = ray.get(args[i])
  self.assertEqual(value[0], i)
```

or randomly choose `100` values and reconstruction them, e.g.,

```python
for _ in range(100):
  i  = np.random.randint(num_objects)
  value = ray.get(args[i])
  self.assertEqual(value[0], i)
```

Actually, maybe add both?